src/Operations.ts

```diff
@@ -7,11 +6,12 @@ export interface DSOperation {
 export class CreateNode implements DSOperation {
     public type = "CreateNode";
     public value: number;
-    public pointer: string;
-    public oldDestination: Node;
+    public id: number;
+    public pointer: AssignPointer;
```

Comment:
The name `pointer` confused me a few times as I was reading through the code, because it's not actually a pointer, it's an operation. I think `assignSuboperation` would be a better name.

Reply:
good idea